pkg/comp-functions/functions/vshnnextcloud/collabora.go

```diff
@@ -0,0 +1,725 @@
+	if err != nil {
+		return runtime.NewWarningResult("Failed to add Collabora SCCs RoleBinding: " + err.Error())
+	}
+
+	// Create coolwsd config map
+	err = createCoolWSDConfigMap(comp, svc)
+	if err != nil {
+		return runtime.NewWarningResult("Failed to add Collabora CoolWSD ConfigMap: " + err.Error())
+	}
+
+	// Create issuer
+	err = createIssuer(comp, svc)
+	if err != nil {
+		return runtime.NewWarningResult("Failed to add Collabora certificate Issuer: " + err.Error())
+	}
+
+	// Create certificate
+	err = createCertificate(comp, svc)
+	if err != nil {
+		return runtime.NewWarningResult("Failed to add Collabora Certificate: " + err.Error())
+	}
+
+	// Create watch-only object
+	err = createWatchOnlyObject(comp, svc)
+	if err != nil {
+		return runtime.NewWarningResult("Failed to add Collabora watch-only Object: " + err.Error())
+	}
+
+	// Create secret with RSA keys
+	err = createSecretWithRSAKeys(comp, svc)
+	if err != nil {
+		return runtime.NewWarningResult("Failed to add Collabora RSA keys: " + err.Error())
+	}
+
+	// copy from watched certificate to new one
+	if svc.Config.Data["isOpenshift"] == "true" {
+		svc.Log.Info("Creating Collabora Secret for Openshift Route")
+		err = createSecretForOpenshiftRoute(comp, svc)
+		if err != nil {
+			svc.Log.Error(err, "Failed to create Collabora Openshift Secret "+err.Error())
+			return runtime.NewWarningResult("Failed to add Collabora Openshift Route: " + err.Error())
+		}
+	}
+
+	svc.Log.Info("Waiting for Nextcloud release to be ready")
+
+	releaseExists := svc.WaitForObservedDependencies(comp.GetName(), comp.GetName()+"-release")
+	if !releaseExists {
+		return runtime.NewWarningResult("Nextcloud instance not yet ready")
+	}
+
+	svc.Log.Info("Nextcloud release is ready, creating STS, Service, Ingress, Issuer, Certificate, ConfigMap, Job")
+
+	// Add Collabora STS
+	err = AddCollaboraSts(comp, svc)
+	if err != nil {
+		return runtime.NewWarningResult("Failed to add Collabora STS: " + err.Error())
+	}
+
+	// Add Collabora service
+	err = AddCollaboraService(comp, svc)
+	if err != nil {
+		return runtime.NewWarningResult("Failed to add Collabora Service: " + err.Error())
+	}
+
+	// Add Collabora ingress
+	err = AddCollaboraIngress(comp, svc)
+	if err != nil {
+		return runtime.NewWarningResult("Failed to add Collabora Ingress: " + err.Error())
+	}
+
+	// Create install Collabora config map
+	err = createInstallCollaboraConfigMap(comp, svc)
+	if err != nil {
+		return runtime.NewWarningResult("Failed to add Collabora ConfigMap: " + err.Error())
+	}
+
+	// Create install Collabora job
+	err = createInstallCollaboraJob(comp, svc)
+	if err != nil {
+		return runtime.NewWarningResult("Failed to add Collabora install Job: " + err.Error())
+	}
+
+	return runtime.NewNormalResult("Collabora deployed and configured!")
+}
+
+func AddCollaboraSts(comp *vshnv1.VSHNNextcloud, svc *runtime.ServiceRuntime) error {
+	res, err := getResources(context.Background(), svc, comp)
```

Comment:
Please don't just pass a `context.Background()` here. We get the proper context in the `DeployCollabora()` function that you can pass here.